src/components/create-post.jsx

```diff
@@ -224,3 +224,47 @@ export default class CreatePost extends React.Component {
     );
   }
 }
+
+/**
+ * Convert 'image/png' blob to 'image/jpeg' blob if:
+ * 1) The PNG size is more than 100 KiB and
+ * 2) JPEG size is less than half the PNG size.
+ *
+ * The returning promise is never failed and returns
+ * either JPEG or the original PNG.
+ *
+ * @param {Blob} blob
+ * @returns {Promise<Blob>}
+ */
+export async function makeJpegIfNeeded(blob) {
+  if (blob.type !== 'image/png' || blob.size < 100 * 1024) {
+    return blob;
+  }
+
+  const src = window.URL.createObjectURL(blob);
+  try {
+    const image = await new Promise((resolve, reject) => {
+      const image = new Image();
+      image.onload = () => resolve(image);
+      image.onerror = () => reject(new Error('Cannot load image'));
+      image.src = src;
+    });
+
+    const canvas = document.createElement('canvas');
+    canvas.width = image.width;
+    canvas.height = image.height;
+    canvas.getContext('2d').drawImage(image, 0, 0);
+
+    const jpeg = await new Promise((resolve) => canvas.toBlob(resolve, "image/jpeg", 0.9));
+    jpeg.name = blob.name.replace(/\.png$/, '.jpg');
+
+    if (jpeg.size < blob.size / 2) {
```

Comment:
Same question about `blob.size` — how big is it in comparison to PNG format? Isn't it much bigger? Or does browser encode it into PNG for us, so size is for PNG?

Reply:
As I say in other answer, yes, it is a PNG file size.